### PR TITLE
Add proxy.config.http.max_proxy_cycles

### PR DESCRIPTION
**Pull request by pbchou**
This commit adds the ability to disable the next hop is self detection
based on the IP address and port. In addition, it allows a configurable
threshold for Via string proxy cycle detection so that it may
be increased from not allowing any cycles to a configurable number of cycles.

### CODE REVIEW

**Comment by pbchou** [comment]
@jrushford this PR is intended to address Issue #7076. Thanks.

File: proxy/http/HttpTransact.cc
Comment (SolidWallOfCode) [inline proxy/http/HttpTransact.cc:6593]: Maybe more like this, for switching to modern C++?
```
std::string_view current { via_field->value_get() };
std::string_view uuid { Machine::instance()->uuid};
std::string_view::size offset;
while (count <= max_proxy_cycles && std::string_view::npos != (offset = current.find(uuid))) {
  current.remove_prefix(offset);
  ++count;
}
```

**Comment by jrushford** [comment]
This passes all unit and regression tests.  I did some functional testing using the configs described in #7076 and all works as expected by setting proxy.config.http.max_proxy_cycles in both records.config and a header_rewrite config.  I'll approve pending approval by @SolidWallOfCode

File: proxy/http/HttpTransact.cc
Comment (pbchou) [reply proxy/http/HttpTransact.cc]: @SolidWallOfCode -- I converted to using C++17 string_view per your comment. Please review as I only read the tutorial for using this class just now. Thanks.

**Comment by pbchou** [comment]
@jrushford -- Can you kick off the checks again? One check failed, but it does not appear to be anything I touched looking at the Jenkins CI output.

**Comment by jrushford** [comment]
[approve ci]

File: proxy/http/HttpTransact.cc
Comment (jrushford) [reply proxy/http/HttpTransact.cc]: [approve ci]

**Comment by pbchou** [comment]
@zwoop -- Can you take a look at the failing clang-analyzer? It says it is in the health check module, which we did not touch.

File: proxy/http/HttpTransact.cc
Comment (pbchou) [reply proxy/http/HttpTransact.cc]: @SolidWallOfCode -- Can you please review? I don't think @jrushford will approve and merge without your input. I pushed another commit to do a static_cast<int> on the string_view::size_type. I did not get compiler warning when compiling as a branch of master, but when I back-ported to our internal 9.0.1, it did complain about int vs long unsigned int. Thanks.

**Comment by pbchou** [comment]
@jrushford -- Do I need to do anything to help this along, or are we just waiting for more approvals?

**Comment by jrushford** [comment]
@pbchou Sorry, I'll review it tomorrow, been busy.  Maybe @SolidWallOfCode could have a look too.

File: proxy/http/HttpTransact.cc
Comment (SolidWallOfCode) [inline proxy/http/HttpTransact.cc:6589]: You could do
```
std::string_view current{via_string};
```
or alternatively
```
std::string_view current{via_field-value_get()};
if ( (count <= max_proxy_cycles) && ! current.empty()) {
  // ....
```

**Review by SolidWallOfCode — APPROVED** [review]

**Review by jrushford — APPROVED** [review]
looks good to me.

**Comment by zwoop** [comment]
Moving out to 9.2.x for now, since we're getting closer to making a 9.1.0 release, this is too big.

**Pull request merged**
